### PR TITLE
Feature/searchbar

### PR DESCRIPTION
**Pull request by Niv-Regev-Moveo**

### CODE REVIEW

File: pokedex/src/components/DetailedCard/DetailedCard.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        </StyledDescription>
+        <StyledStatsDetails>
+          <div>
+            <h3>Stats</h3>
+            <p>HP: {pokemon.stats[0].base_stat}</p>
+            <p>Attack: {pokemon.stats[1].base_stat}</p>
+            <p>Defense: {pokemon.stats[2].base_stat}</p>
+          </div>
+          <div>
+            <br></br>
```
Comment (yuvalkom-M) [inline pokedex/src/components/DetailedCard/DetailedCard.tsx]: Less common to use </br>. You can give margin-top if needed

File: pokedex/src/components/FavoritePage/FavoritePage.tsx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const FavoritePage = () => {
+  return <h1>hello favorite page</h1>;
```
Comment (yuvalkom-M) [inline pokedex/src/components/FavoritePage/FavoritePage.tsx]: remove unnecessary files (all files related to favorite page)

File: pokedex/src/components/HomePage/HomePage.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import PageHeader from "../PageHeader/PageHeader";
+import PokeSearchResult from "../PokeSearchResults/PokeSearchResults";
+import PokemonCard from "../Card/Card";
+import { StyledCardsContainer } from "../PokemonList/styledPokemonList";
+import { getPokemons, getSpecificPokemon, BASE_URL } from "../../utils/api";
+
+function HomePage() {
+  const [pokemonsData, setPokemonsData] = useState<any[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [url, setUrl] = useState(BASE_URL);
```
Comment (yuvalkom-M) [inline pokedex/src/components/HomePage/HomePage.tsx:19]: why use useState for url? (no use of setUrl)

File: pokedex/src/components/PokeSearchResults/PokeSearchResults.tsx [hunk 4]
```diff
@@ -1,12 +1,51 @@
...
+  StyledSearchOption,
+  StyledSearchBarDiv,
+} from "./styledPokeSearchResults";
+
 const PokeSearchResult = ({ pokemonsData }: any) => {
+  const [selectedPokemon, setSelectedPokemon] = useState<any>(null);
+
+  const onSelect = (selectedName: string) => {
+    const selected = pokemonsData.find(
+      (pokemon: any) => pokemon.name === selectedName
```
Comment (yuvalkom-M) [inline pokedex/src/components/PokeSearchResults/PokeSearchResults.tsx]: don't use any as type

File: pokedex/src/components/PokeSearchResults/PokeSearchResults.tsx [hunk 5]
```diff
@@ -1,12 +1,51 @@
...
+    );
+    setSelectedPokemon(selected);
+  };
+
   return (
     <div>
-      <h2>Search Results</h2>
-      <ul>
-        {pokemonsData.map((result: any, id: any) => (
```
Comment (yuvalkom-M) [inline pokedex/src/components/PokeSearchResults/PokeSearchResults.tsx:6]: don't use any as type

File: pokedex/src/components/PokeSearchResults/PokeSearchResults.tsx [hunk 6]
```diff
@@ -1,12 +1,51 @@
...
-      </ul>
+      <StyledSearchBarDiv>
+        <StyledSearchBar
+          defaultValue=""
+          onChange={(e) => onSelect(e.target.value)}
+        >
+          <option value="" disabled>
+            Select a Pokémon
+          </option>
+          {pokemonsData.map((result: any, id: any) => (
```
Comment (yuvalkom-M) [inline pokedex/src/components/PokeSearchResults/PokeSearchResults.tsx]: same don't use any as type

File: pokedex/src/components/PokemonList/Pokemonlist.tsx [hunk 7]
```diff
@@ -1,6 +1,14 @@
 import PokemonCard from "../Card/Card";
 
-const PokemonList = ({ pokemonsData, loading }: any) => {
+type typePokemonListProps = {
+  pokemonsData: any[];
```
Comment (yuvalkom-M) [inline pokedex/src/components/PokemonList/Pokemonlist.tsx]: don't use any as type

File: pokedex/src/utils/api.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    const res: AxiosResponse<{ results: Pokemon[] }> = await axios.get(url);
+
+    return res.data.results;
+  } catch (error) {
+    console.error("Error fetching data:", error);
+    throw error;
+  }
+}
+
+export async function getSpecificPokemon(urls: string[]): Promise<any[]> {
```
Comment (yuvalkom-M) [inline pokedex/src/utils/api.ts]: don't use any as type

File: pokedex/src/components/NavBar/NavBar.tsx [hunk 9]
```diff
@@ -10,19 +15,41 @@ type NavBarProps = {
...
-          <a href={hrefHomePage}>{homePageName}</a>
-        </li>
-        <li>
-          <a href={hrefFavoritePage}>{favoritePageName}</a>
-        </li>
-      </ul>
+      <StyledPagesLinks>
+        <StyledLinkToPages
+          isActive={activeLink === hrefHomePage}
+          onClick={() => handleLinkClick(hrefHomePage)}
```
Comment (yuvalkom-M) [inline pokedex/src/components/NavBar/NavBar.tsx]: do you need twice onClick for the same nav item?

**Review by OmerAMoveo — APPROVED** [review]

**Pull request merged**